cpp/util/read_key.cc

```diff
@@ -56,4 +68,36 @@ util::StatusOr<EVP_PKEY*> ReadPublicKey(const std::string& file) {
 }
 
 
+util::StatusOr<EVP_PKEY*> ReadEnginePrivateKey(const std::string& file, const std::string& engine_name) {
+  unique_ptr<FILE, void (*)(FILE*)> fp(fopen(file.c_str(), "r"), FileCloser);
+
+  if (!fp)
+    return util::Status(util::error::NOT_FOUND, "key file not found: " + file);
+  OPENSSL_config(NULL);
+  ENGINE_load_dynamic();
+  atexit(ENGINE_cleanup);
+
+
+  engine = ENGINE_by_id(engine_name.c_str());
+  if (!engine)
+    return util::Status(util::error::NOT_FOUND, "engine not found: " + engine_name);
+  atexit(EngineFree);
+
+  if (!ENGINE_init(engine)) {
+    static char buf[1024];
+    ERR_error_string(ERR_get_error(), buf);
+    return util::Status(util::error::ENGINE_INIT_FAILED, "engine init failed: " + std::string(buf));
```

Comment:
Please choose one of the existing codes in `status.h`, `FAILED_PRECONDITION` would be fine I think (the idea is to cover all conditions with a finite number of status codes, which makes it easier to handle for callers.)